src/assets/DockerCompose/docker_compose.md

```diff
@@ -0,0 +1,211 @@
+---
+title: Containerize a Web Application using docker compose
+authorName: Sagar Gurung
+authorAvatar: https://avatars.githubusercontent.com/u/46086950?v=4
+authorLink: https://github.com/SagarGi
+createdAt: Aug 31, 2023
+tags: docker-compose, docker, containerization, yaml, services
+banner: https://blog.jankaritech.com/src/assets/DockerCompose/images/docker-cmopose-background.png
+---
+
+In this blog we will basically be containerizing a web application consisting of `MySQL`, `phpmyadmin`, `frontend` and `backend` as service using docker compose.
+Also, briefly get into how all those services are running and how multiple services make up an application running in multiple containers
+> Note: This article does not provide concepts of docker. To know concepts of docker [visit here.](https://blog.jankaritech.com/#/blog/How%20to%20use%20selenium%20in%20docker%3F)
+
+## Brief About Docker Compose
+In very short a docker compose helps to define and run docker applications in multiple containers. With docker compose we can define a `yml` file where we can define our services, networks, volumes required for our application to get it running. Basically configuring, defining and managing all the necessary things required for our application becomes easy with docker cmopose.
+
+And in this blog we will also be defining and configuring simple services for our application through a compose file and run it using docker compse.
+
+## What is the application about (to be run with docker compose)?
+The application we will be running with docker compose has mainly 3 services:
+  1. `frontend` service
+  2. `backend` service
+  3. `mysql service` (database service)
+
+Also, we will be running the 4th service `phpmyadmin` which is used for managing and administering the MySQL through GUI.
+This is a very simple application for demo, where in `mysql` database there are lists of user information and through the `frontend` service we will request to `backend` service to get all those users and to be listed in our application UI. Like this all those application services are dependent and collaborating to each other. We will be configuring the exact same thing through docker compose.
+
+## Prerequisites
+Before we start, lets make sure following things get installed into your system.
+  1. install [docker](https://www.digitalocean.com/community/tutorials/how-to-install-and-use-docker-on-ubuntu-20-04)
+  2. install [docker compose](https://www.digitalocean.com/community/tutorials/how-to-install-and-use-docker-compose-on-ubuntu-20-04)
+  > Note: I will be doing the whole setup with Ubuntu 22.04 LTS. Things might differ if you have different operating system.
+
+## Getting Started
+
+Since we have enough information about the application And, also I assumed that the prerequisites are fulfilled. we can now jump to the section for defining and configuring our application with docker compose.
```

Comment:
```suggestion
Since we have enough information about the application and the prerequisites are fulfilled. we can now jump to the section for defining and configuring our application with docker compose.
```